### ISSUE TITLE
No error regarding shorthand-property-no-redundant-values

### ISSUE THREAD

**Issue by maciejmyslinski** [stylelint/stylelint#2542]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where `shorthand-property-no-redundant-values` does not produce error while it should.

> Which rule, if any, is this issue related to?

`shorthand-property-no-redundant-values`

> What CSS is needed to reproduce this issue?

e.g.

```scss
.block {
  &__element {
    margin: 0 0 $vertical-rhythm * 2 - 43px 0;
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```js
module.exports = {
  "rules": {
    "shorthand-property-no-redundant-values": true
  }
}
```

> Which version of stylelint are you using?

`7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint "**/*.scss" --syntax scss --config ".stylelintrc.js"` and VS code stylelint extension. Both work the same.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS.

> What did you expect to happen?

Warning to be shown

> What actually happened (e.g. what warnings or errors you are getting)?

No warnings.

**Comment by hudochenkov** [comment]
Thank you for using template!

Stylelint doesn't compile Sass, so it doesn't know that Sass mathematical equation is one value. This is why it doesn't show any warnings.

**Comment by evilebottnawi** [comment]
@hudochenkov hm, i think i can do this rule to works with variables :nerd_face:

**Comment by jeddy3** [comment]
I think the rule correctly ignores the non-standard syntax (i.e. operators outside of a `calc()` function). It's consistent with our treatment of non-standard syntax everywhere else.

> hm, i think i can do this rule to works with variables

I think such a feature should be a plugin within `stylelint-scss`, perhaps a plugin that possible builds on top of this rule.

**Comment by hudochenkov** [comment]
> I think the rule correctly ignores the non-standard syntax (i.e. operators outside of a calc() function).

The rule [ignores](https://github.com/stylelint/stylelint/blob/cbcb4a2706104f1dcc6601b87938c6bee4834461/lib/rules/shorthand-property-no-redundant-values/__tests__/index.js#L55-L62) `calc`, other functions, and custom properties.

**Comment by evilebottnawi** [comment]
@jeddy3 i agree about `sass/scss/less`, but `custom-variables`, `calc` and other stuff is part of `css` and i think rule should handles their.

**Comment by evilebottnawi** [comment]
@jeddy3 maybe we should create separate issue about support official `css` (`css-variables`, `calc` and etc) in this rule and add milestone `future major`? Maybe I could even do it to version 8

**Comment by jeddy3** [comment]
> but custom-variables, calc and other stuff is part of css and i think rule should handles their.
> maybe we should create separate issue about support official

Yes, a separate issue. This particular request is for this non-standard syntax: `margin: 0 0 $vertical-rhythm * 2 - 43px 0;`.  Which is outside of the scope of core rules.

**Comment by gnarmedia** [comment]
I think due to ignoring this rule negative values are also ignored? I am seeing this in my current stylesheet:
    margin: 0 -2% 0 -2%;
no error

**Comment by hudochenkov** [comment]
@gnarmedia please, create a new issue if you noticed problems with stylelint.
